import "Ws.proto";
import "Identifiers.proto";

package Opal;

option java_package = "org.obiba.opal.web.model";

message Message {
  required string msg = 1;
  required int64 timestamp = 2;
}

message CommandStateDto {
  enum Status {
    NOT_STARTED = 0;
    IN_PROGRESS = 1;
    SUCCEEDED = 2;
    FAILED = 3;
    CANCEL_PENDING = 4;
    CANCELED = 5;
  }

  message ProgressDto {
    required string message = 1;
    required uint32 current = 2;
    required uint32 end = 3;
    required uint32 percent = 4;
  }

  required uint32 id = 1;
  required string command = 2;
  required string commandArgs = 3;
  required string owner = 4;
  required string status = 5;
  optional string startTime = 6;
  optional string endTime = 7;
  repeated Message messages = 8;
  optional string project = 9;
  required string name = 10;
  optional ProgressDto progress = 11;
}

message ImportCommandOptionsDto {
  optional string destination = 2;
  optional string archive = 3;
  repeated string files = 4;
  optional string source = 5;
  repeated string tables = 7;
  optional bool incremental = 9 [default = false];
  optional Identifiers.IdentifiersMappingConfigDto idConfig = 10;
  optional bool createVariables = 11 [default = true];
}

message CopyCommandOptionsDto {
  optional string source = 1;
  required string destination = 2;
  optional bool nonIncremental = 3;
  optional bool noValues = 4;
  optional bool noVariables = 5;
  repeated string tables = 6;
  optional bool copyNullValues = 7;
  optional string destinationTableName = 8;
  optional string query = 9;
}

message ExportCommandOptionsDto {
  optional string source = 1;
  optional string destination = 2;
  optional string out = 3;
  optional bool nonIncremental = 4;
  optional bool noValues = 5;
  optional bool noVariables = 6;
  repeated string tables = 7;
  optional string format = 8;
  optional bool copyNullValues = 9;
  optional string destinationTableName = 10;
  optional Identifiers.IdentifiersMappingConfigDto idConfig = 11;
  optional string query = 12;
}

message ReportCommandOptionsDto {
  required string name = 1;
  required string project = 2;
}